core-web/libs/sdk/client/src/lib/sdk-js-client.ts

```diff
@@ -0,0 +1,278 @@
+     * @example
+     * `1` (or unspecified) returns content in the default language of the site.
+     *
+     * @link https://www.dotcms.com/docs/latest/system-language-properties#DefaultLanguage
+     * @link https://www.dotcms.com/docs/latest/adding-and-editing-languages#LanguageID
+     * @type {number}
+     */
+    languageId?: number;
+};
+
+function isValidUrl(url: string): boolean {
+    try {
+        new URL(url);
+
+        return true;
+    } catch (error) {
+        return false;
+    }
+}
+
+/**
+ * `DotCmsClient` is a TypeScript class that provides methods to interact with the DotCMS REST API.
+ * It requires a configuration object on instantiation, which includes the DotCMS URL, site ID, and authentication token.
+ *
+ * @class DotCmsClient
+ *
+ * @property {ClientConfig} config - The configuration object for the DotCMS client.
+ *
+ * @method constructor(config: ClientConfig) - Constructs a new instance of the DotCmsClient class.
+ *
+ * @method getPage(options: PageApiOptions): Promise<unknown> - Retrieves all the elements of any Page in your dotCMS system in JSON format.
+ *
+ * @method getNav(options: NavApiOptions = { depth: 0, path: '/', languageId: 1 }): Promise<unknown> - Retrieves information about the dotCMS file and folder tree.
+ *
+ */
+export class DotCmsClient {
+    private config: ClientConfig;
+
+    constructor(config: ClientConfig) {
+        if (!config.dotcmsUrl) {
+            throw new Error("Invalid configuration - 'dotcmsUrl' is required");
+        }
+
+        if (!isValidUrl(config.dotcmsUrl)) {
+            throw new Error("Invalid configuration - 'dotcmsUrl' must be a valid URL");
+        }
+
+        if (!config.authToken) {
+            throw new Error("Invalid configuration - 'authToken' is required");
+        }
+
+        this.config = config;
+    }
+
+    /**
+     * `getPage` is an asynchronous method of the `DotCmsClient` class that retrieves all the elements of any Page in your dotCMS system in JSON format.
+     * It takes a `PageApiOptions` object as a parameter and returns a Promise that resolves to the response from the DotCMS API.
+     *
+     * The Page API enables you to retrieve all the elements of any Page in your dotCMS system.
+     * The elements may be retrieved in JSON format.
+     *
+     * @link https://www.dotcms.com/docs/latest/page-rest-api-layout-as-a-service-laas
+     * @method getPage
+     * @async
+     * @param {PageApiOptions} options - The options for the Page API call.
+     * @returns {Promise<unknown>} - A Promise that resolves to the response from the DotCMS API.
+     * @throws {Error} - Throws an error if the options are not valid.
+     */
+    async getPage(options: PageApiOptions): Promise<unknown> {
+        this.validatePageOptions(options);
+
+        const queryParamsObj: Record<string, string> = {};
+        for (const [key, value] of Object.entries(options)) {
+            if (value !== undefined && key !== 'path') {
```

Comment:
Wonder if we can remove at least one _if_ here 

if (value !== undefined && key !== 'path' && key !== 'siteId') {
    if (key === 'personaId') {
        queryParamsObj['com.dotmarketing.persona.id'] = String(value);
    } else {
        queryParamsObj[key] = String(value);
    }
}